### ISSUE TITLE
SingleChoiceAnswerFormat default selection does not work

### ISSUE THREAD

**Issue by SaqibHussain44** [quickbirdstudios/survey_kit#27]
The survey does not preselect the default answer when I provide the defaultSelection.

Here is the code snippet

```
this.questions.asMap().forEach((index, each) => {
  choices = [],
  each['options'].forEach((eachOption) {
    choices.add(TextChoice(
        text: eachOption['text'],
        value: eachOption['text']
      )
    );
  }),
  steps.add(QuestionStep(
    isOptional: false,
    id: StepIdentifier(id: index.toString()),
    title: 'Question ' + index.toString(),
    text: each['question'],
    answerFormat: SingleChoiceAnswerFormat(
      defaultSelection: TextChoice(
        text: capitalize(getDefaultSelection(each['question'])),
        value: getDefaultSelection(each['question'])
      ),
      textChoices: choices
    )
  ))
});
```

**Comment by sohail0992** [comment]
@Numoy can you please check this one also?

**Comment by Numoy** [comment]
Fixed in version 0.0.12

**Comment by sohail0992** [comment]
@Numoy  Update the package to version 0.0.12,  constructing the step like below, still on pressing next without any selection, next question appear.
      ```
  steps.add(QuestionStep(
              stepIdentifier: StepIdentifier(id: index.toString()),
              title: 'Question ' + index.toString() + '/' + total.toString(),
              text: each['question'],
              answerFormat: SingleChoiceAnswerFormat(
                  defaultSelection: isAlreadySubmitted
                      ? TextChoice(
                          text:
                              capitalize(getDefaultSelection(each['question'])),
                          value: getDefaultSelection(each['question']))
                      : null,
                  textChoices: choices))),
        });
```
